agent/agentbootstrap/bootstrap.go

```diff
@@ -141,7 +141,8 @@ func InitializeState(
 		modelType = state.ModelTypeCAAS
 	}
 	ctrl, err := state.Initialize(state.InitializeParams{
-		Clock: clock.WallClock,
+		SSHServerHostKey: args.SSHServerHostKey,
```

Comment:
You could simplify the patch change by not adding this to the initialisation params.

Where you added the getter in _state/controller.go_, you could add a setter also. In this method you could then just call the setter directly on the state that is acquired at line 182 below.

However, that leaves a setter that we would have to ensure is never recruited anywhere but bootstrap.

In that light, I'm OK with this change. All this has to go as part of the Dqlite migration anyway.

Reply:
Sure thing, I'll leave as is then. I'm not 100% sure I'm following you in this comment, but if this is OK I'll leave it.